src/endpoints/edit.rs

```diff
@@ -457,8 +457,8 @@ pub async fn post_edit(
                             header::LOCATION,
                             format!(
                                 "{}/upload/{}",
-                                ARGS.public_path_as_str(),
-                                pastas[i].id_as_animals()
+                                data.args.public_path_as_str(),
+                                pastas[i].id_as_animals(&data.args)
```

Comment:
might make sense to pass the most specific thing possible to each function. In this case `id_as_animals` needs `hash_id`. So `data.args.hash_id` could be the argument. Now that is a rather long argument, but it should get shorter as we apply struct unpacking in the argument to this function (see earlier comment).